### ISSUE TITLE
Prevent using SET NULL constraints in self-referencing relationships

### ISSUE THREAD

**Issue by jrvidotti** [directus/directus#7920]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

Creating a M2O self referencing column throws the following error:
```
{
  "message": "ALTER TABLE [cargo] ADD CONSTRAINT [cargo_cargo_lideranca_foreign] FOREIGN KEY ([cargo_lideranca]) REFERENCES [cargo] ([id]) ON DELETE SET NULL - Could not create constraint or index. See previous errors.",
  "extensions": {
	"code": "INTERNAL_SERVER_ERROR"
  }
}
```
If I refresh the page, the column is created. Trying to run command directly in SQL Server, throws the error:
```
Mensagem 1785, Nível 16, Estado 0, Linha 1
Introducing FOREIGN KEY constraint 'cargo_cargo_lideranca_foreign' on table 'cargo' may cause cycles or multiple cascade paths. Specify ON DELETE NO ACTION or ON UPDATE NO ACTION, or modify other FOREIGN KEY constraints.
Mensagem 1750, Nível 16, Estado 1, Linha 1
Could not create constraint or index. See previous errors.
```
Changing command ON DELETE to NO ACTION works!

### To Reproduce

1. Create a new Collection
2. Create a M2O Field referencing the same collection

### What version of Directus are you using?

v9.0.0-rc.92

### What version of Node.js are you using?

v16.8.0

### What database are you using?

SQL Server v15.0.4231.1

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu WSL2

### How are you deploying Directus?

local

**Comment by rijkvanzanten** [comment]
This is not necessarily a bug we can solve in Directus, but rather a limitation of the database 🤔 That being said, we can improve the setup by not defaulting to NOT NULL when you have a self-referencing relationship (and potentially disabling the option, as it's unreliable in most databases)

**Comment by azrikahar** [comment]
This should be resolved by #9109, particularly the following implementations in each of the relationships:

https://github.com/directus/directus/blob/5d23dfd37f8b75039c0759820b0373a7ba459f97/app/src/modules/settings/routes/data-model/field-detail/store/alterations/m2a.ts#L149-L153

https://github.com/directus/directus/blob/5d23dfd37f8b75039c0759820b0373a7ba459f97/app/src/modules/settings/routes/data-model/field-detail/store/alterations/m2m.ts#L104-L112

https://github.com/directus/directus/blob/5d23dfd37f8b75039c0759820b0373a7ba459f97/app/src/modules/settings/routes/data-model/field-detail/store/alterations/m2o.ts#L82-L90

https://github.com/directus/directus/blob/5d23dfd37f8b75039c0759820b0373a7ba459f97/app/src/modules/settings/routes/data-model/field-detail/store/alterations/o2m.ts#L61-L69
